repos/system_upgrade/el7toel8/actors/checkfstab/actor.py

```diff
@@ -0,0 +1,81 @@
+import os
+import time
+
+from leapp.actors import Actor
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.common.fstab import drop_xfs_options
+from leapp.libraries.stdlib import api
+from leapp.models import Report, FstabSignal
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+from leapp import reporting
+
+
+PREFACE = """# Below are the modified contents of /etc/fstab that have been generated for
+# the upgraded system. Changes to this file will not be reflected. If you
+# believe the file has been modified incorrectly, please report a bug and
+# attach this file and the current /etc/ftsab.
+
+# Generated by leapp on {}
+# ========================================
+"""
+FSTAB_LOGFILE = '/var/log/leapp/fstab.new'
+
+
+class CheckFstab(Actor):
+    """
+    Scan /etc/fstab for XFS mount options that have been removed in RHEL 8.
+
+    In case one or more of the options are present in /etc/fstab, generate new contents of the file
+    for the target system and provide it to the user via /var/log/leapp/fstab.new for inspection.
+    """
+
+    name = 'check_fstab'
+    consumes = ()
+    produces = (Report, FstabSignal)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        fstab_old = []
+        try:
+            with open('/etc/fstab') as f:
+                fstab_old = f.readlines()
+        except (IOError, OSError):
+            raise StopActorExecutionError(message='Could not open /etc/fstab for reading')
+
+        fstab_new = drop_xfs_options(fstab_old)
+        if fstab_new == fstab_old:
+            api.current_logger().debug('No changes made in /etc/fstab')
+            if os.path.exists(FSTAB_LOGFILE):
+                api.current_logger().debug('{} exists from a previous run, cleaning up'.format(FSTAB_LOGFILE))
+                os.remove(FSTAB_LOGFILE)
+            return
+
+        try:
+            with open(FSTAB_LOGFILE, 'w') as f:
+                f.write(PREFACE.format(time.ctime()))
+                f.writelines(fstab_new)
+        except (IOError, OSError):
+            raise StopActorExecutionError(message='Could not open {} for writing'.format(FSTAB_LOGFILE))
+
+        reporting.create_report([
+            reporting.Title(
+                '/etc/fstab will be modified during upgrade'
+            ),
+            reporting.Summary(
+                'In RHEL 8, several mount options for XFS are removed and their usage in /etc/fstab would '
+                'result in an unbootable system. Since the options are already deprecated and do not have '
+                'any effect on the mounts, leapp will remove them.\n'
+                'A copy of the new /etc/fstab has been written to {}. '.format(FSTAB_LOGFILE) +
+                'Changes to this file will not be reflected in the upgrade process.'
+            ),
```

Comment:
I am missing the instruction (maybe remediation hint?) people should take a look into the file  to check it's correct, or they can update the file manually to drop the deprecated/removed options already and re-run leapp.